extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/VertxProducer.java

```diff
@@ -115,6 +110,8 @@ void undeployVerticles(@Observes @BeforeDestroyed(ApplicationScoped.class) Objec
                     // Only existing instances are considered
                     try {
                         AbstractVerticle verticle = (AbstractVerticle) instance;
+                        io.vertx.mutiny.core.Vertx mutiny = beanManager.createInstance()
```

Comment:
So the only difference compared to the injected observer param is that the `Vertx` instance is only obtained if there is at least one bean that extends `AbstractVerticle`?

Reply:
One already created bean, which generally won't happen until runtime when you actually have vert.x. 

Currently if you fail in static init then this method is called, but vert.x is not ready yet (as it is a runtime bean), so you get an extra unrelated exception in your logs (and possible unclean shutdown).

Reply:
I see. That makes sense.